ibis/expr/tests/test_analysis.py

```diff
@@ -301,3 +301,14 @@ def test_join_table_choice():
     t = x.aggregate(cnt=x.n.count())
     predicate = t.cnt > 0
     assert L.sub_for(predicate, [(t, t.op().table)]).equals(predicate)
+
+
+def test_is_ancestor_analytic():
+    x = ibis.table(ibis.schema([('col', 'int32')]), 'x')
+    with_filter_col = x[x.columns + [ibis.null().name('filter')]]
+    filtered = with_filter_col[with_filter_col['filter'].isnull()]
+    subquery = filtered[filtered.columns]
+
+    with_analytic = subquery[subquery.columns + [subquery.count().name('analytic')]]
+
+    assert not subquery.op().is_ancestor(with_analytic)
```

Comment:
I think the semantics of `is_ancestor` are the other way around, so this should be `False` even if the code path had been running before this PR.

The ancestor relations of the `self` argument are traversed until an ancestor that's equivalent to the passed argument is reached or there are no more ancestors.

So, I would expect this assertion to be valid.

However, I would expect the following assertion to also be valid, which I don't think it is:

```python
with_analytic.op().is_ancestor(subquery)
```
since `with_analytic` is derived from `subquery`.